src/sprout/Helpers/WorkerFilesBackendMigrate.php

```diff
@@ -0,0 +1,373 @@
+
+        $file_model->filesize = $filesize;
+        $file_model->imagesize = json_encode($imagesize);
+
+        $file_model->filename_migrated = $new_filename;
+        $file_model->backend_migrated = $this->_new_backend_type;
+        $file_model->date_migrated = Pdb::now();
+
+        $file_model->date_added = $file_model->date_added ? $file_model->date_added : $now;
+        $file_model->date_modified = $file_model->date_modified ? $file_model->date_modified : $now;
+        $file_model->date_file_modified = $file_model->date_file_modified ? $file_model->date_file_modified : $now; // We don't know
+
+        $is_orphan = empty($file_model->id);
+
+        $res = $file_model->save();
+
+        if (!$res) {
+            Worker::message("Failed to save file record for file #{$file_model->id} | {$filename}");
+            return $res;
+        }
+
+        Worker::message("Saved file record for file #{$file_model->id} | {$filename}");
+
+        // Only 'files' can live in categories
+        if (!$file_model instanceof FileModel) {
+            return true;
+        }
+
+        // Add any bulk categories if set
+        if (!$is_orphan and !empty($this->_options['category_id_files'])) {
+            $res = Category::insertInto('files', $file_model->id, $this->_options['category_id_files']);
+            Worker::message("Adding to file category #{$this->_options['category_id_files']} | " . ($res ? 'OK' : 'FAIL'));
+        }
+
+        if ($is_orphan and !empty($this->_options['category_id_orphans'])) {
+            Category::insertInto('files', $file_model->id, $this->_options['category_id_orphans']);
+            Worker::message("Adding to orphan category #{$this->_options['category_id_files']} | " . ($res ? 'OK' : 'FAIL'));
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Big sweeping change to move all migrated data into active cols
+     *
+     * @return void
+     */
+    private function updateDatabaseRecords()
+    {
+        $tables = [
+            'files',
+            'file_transforms',
+        ];
+
+        foreach ($tables as $table) {
+            $q = "UPDATE ~{$table}
+                SET backend_type = backend_migrated,
+                    filename = filename_migrated,
+                    backend_migrated = null,
+                    filename_migrated = null,
+                    date_migrated = null
+                WHERE backend_migrated = ?
+                AND date_migrated IS NOT NULL";
+            $params = [$this->_new_backend_type, $this->_new_backend_type];
+
+            $num_updated = Pdb::query($q, $params, 'count');
+            Worker::message("Updated {$num_updated} rows for files with backend_migrated");
+        }
+    }
+
+
+    /**
+     * Extract a transform name from a filename string, if one exists
+     *
+     * @param string $filename
+     *
+     * @return int|string|null
+     */
+    private function getTransformName(string $filename)
```

Comment:
Would this have any use outside of this worker? Could go into the FileTransform helper.

Reply:
I can see it being useful for sure. I'm always keen on having a good set of helpers on hand to... help